src/linear_algebra/matrix.lean

```diff
@@ -998,6 +1029,90 @@ end reindexing
 
 end matrix
 
+namespace algebra
+
+section lmul
+
+variables {R S T : Type*} [comm_ring R] [comm_ring S] [comm_ring T]
+variables [algebra R S] [algebra S T] [algebra R T] [is_scalar_tower R S T]
+variables {m n : Type*} [fintype m] [decidable_eq m] [fintype n] [decidable_eq n]
+variables {b : m → S} (hb : is_basis R b) {c : n → T} (hc : is_basis S c)
+
+open algebra
+
+@[simp] lemma lmul_algebra_map (x : R) :
+  lmul R S (algebra_map R S x) = algebra.lsmul R S x :=
+linear_map.ext (λ s, by simp [smul_def''])
+
+lemma to_matrix_lmul' (x : S) (i j) :
+  linear_map.to_matrix hb hb (lmul R S x) i j = hb.repr (x * b j) i :=
+by rw [linear_map.to_matrix_apply', lmul_apply]
+
+@[simp] lemma to_matrix_lsmul (x : R) (i j) :
+  linear_map.to_matrix hb hb (algebra.lsmul R S x) i j = if i = j then x else 0 :=
+by { rw [linear_map.to_matrix_apply', algebra.lsmul_coe, linear_map.map_smul, finsupp.smul_apply,
+         hb.repr_self_apply, smul_eq_mul, mul_boole],
+     congr' 1; simp only [eq_comm] }
+
+/-- `left_mul_matrix hb x` is the matrix corresponding to the linear map `λ y, x * y`
+
+`left_mul_matrix_eq_repr_mul` gives a formula for the entries of `left_mul_matrix`.
+
+This definition is useful for doing (more) explicit computations with `algebra.lmul`,
+such as the trace form or norm map for algebras.
+-/
+noncomputable def left_mul_matrix : S →ₐ[R] matrix m m R :=
+{ to_fun := λ x, linear_map.to_matrix hb hb (algebra.lmul R S x),
+  map_zero' := by rw [alg_hom.map_zero, linear_equiv.map_zero],
+  map_one' := by rw [alg_hom.map_one, linear_map.to_matrix_one],
+  map_add' := λ x y, by rw [alg_hom.map_add, linear_equiv.map_add],
+  map_mul' := λ x y, by rw [alg_hom.map_mul, linear_map.to_matrix_mul, matrix.mul_eq_mul],
+  commutes' := λ r, by { ext, rw [lmul_algebra_map, to_matrix_lsmul,
+                                  algebra_map_matrix_apply, id.map_eq_self] } }
+
+lemma left_mul_matrix_apply (x : S) :
+  left_mul_matrix hb x = linear_map.to_matrix hb hb (lmul R S x) := rfl
+
+lemma left_mul_matrix_eq_repr_mul (x : S) (i j) :
+  left_mul_matrix hb x i j = hb.repr (x * b j) i :=
+by rw [left_mul_matrix_apply, linear_map.to_matrix_apply, algebra.lmul_apply, hb.equiv_fun_apply]
```

Comment:
I think the statement I'm looking for might be something like
```suggestion
lemma left_mul_matrix_eq_repr_mul (x y : S) (i j) :
  hb.equiv_fun.symm ((left_mul_matrix hb x).mul_vec (hb.repr y)) = x * y := sorry
```
There might be a better way to spell that, but it makes clear exactly how to recover the muliply that this function claims to perform!

Reply:
I suppose `(left_mul_matrix hb x).mul_vec (hb.repr y) = hb.repr (x * y)` is probably just as good, and is easier to come up with a name for, `left_mul_matrix_mul_vec_repr`.